pinot-common/src/main/java/org/apache/pinot/common/request/context/RequestContextUtils.java

```diff
@@ -163,94 +165,113 @@ public static FunctionContext getFunction(FunctionCallAstNode astNode) {
+                  RangePredicate.UNBOUNDED));
+        case GREATER_THAN_OR_EQUAL:
+          return new FilterContext(FilterContext.Type.PREDICATE, null,
+              new RangePredicate(getExpression(operands.get(0)), true, getStringValue(operands.get(1)), false,
+                  RangePredicate.UNBOUNDED));
+        case LESS_THAN:
+          return new FilterContext(FilterContext.Type.PREDICATE, null,
+              new RangePredicate(getExpression(operands.get(0)), false, RangePredicate.UNBOUNDED, false,
+                  getStringValue(operands.get(1))));
+        case LESS_THAN_OR_EQUAL:
+          return new FilterContext(FilterContext.Type.PREDICATE, null,
+              new RangePredicate(getExpression(operands.get(0)), false, RangePredicate.UNBOUNDED, true,
+                  getStringValue(operands.get(1))));
+        case BETWEEN:
+          return new FilterContext(FilterContext.Type.PREDICATE, null,
+              new RangePredicate(getExpression(operands.get(0)), true, getStringValue(operands.get(1)), true,
+                  getStringValue(operands.get(2))));
+        case RANGE:
+          return new FilterContext(FilterContext.Type.PREDICATE, null,
+              new RangePredicate(getExpression(operands.get(0)), getStringValue(operands.get(1))));
+        case REGEXP_LIKE:
+          return new FilterContext(FilterContext.Type.PREDICATE, null,
+              new RegexpLikePredicate(getExpression(operands.get(0)), getStringValue(operands.get(1))));
+        case LIKE:
+          return new FilterContext(FilterContext.Type.PREDICATE, null,
+              new RegexpLikePredicate(getExpression(operands.get(0)),
+                  RegexpPatternConverterUtils.likeToRegexpLike(getStringValue(operands.get(1)))));
+        case TEXT_MATCH:
+          return new FilterContext(FilterContext.Type.PREDICATE, null,
+              new TextMatchPredicate(getExpression(operands.get(0)), getStringValue(operands.get(1))));
+        case JSON_MATCH:
+          return new FilterContext(FilterContext.Type.PREDICATE, null,
+              new JsonMatchPredicate(getExpression(operands.get(0)), getStringValue(operands.get(1))));
+        case IS_NULL:
+          return new FilterContext(FilterContext.Type.PREDICATE, null,
+              new IsNullPredicate(getExpression(operands.get(0))));
+        case IS_NOT_NULL:
+          return new FilterContext(FilterContext.Type.PREDICATE, null,
+              new IsNotNullPredicate(getExpression(operands.get(0))));
+        default:
+          throw new IllegalStateException();
+      }
+    } else if (type == ExpressionType.IDENTIFIER) {
+      // Convert "WHERE a" to "WHERE a = true"
+      return new FilterContext(FilterContext.Type.PREDICATE, null,
+          new EqPredicate(getExpression(thriftExpression), getStringValue(RequestUtils.getLiteralExpression(true))));
     }
+
+    throw new IllegalArgumentException();
```

Comment:
We should also handle `literal`, where we handle `false`, `0`, `0.0` as false, others as true. Currently literal is always represented as string, so we have to take this work-around. Once we preserve the literal type, this can be simplified

Reply:
Can we add support for literals as a separate PR? (asking because adding support and test for literals will make this PR larger). I'll send out a PR just for literals soon.